creator-node/src/blacklistManager.js

```diff
@@ -116,19 +110,61 @@ class BlacklistManager {
       await this.addToRedis(REDIS_SET_BLACKLIST_USERID_KEY, userIdsToBlacklist)
       await this.addToRedis(
         REDIS_SET_BLACKLIST_SEGMENTCID_KEY,
-        segmentCIDsToBlacklist
-      )
-
-      // Creates a mapping of CID to blacklisted trackIds in redis. This should not include the trackIds from users.
-      await this.addToRedis(
-        REDIS_MAP_BLACKLIST_SEGMENTCID_TO_TRACKID_KEY,
-        trackIdToSegments
+        segmentsToBlacklist
       )
     } catch (e) {
       throw new Error(
-        `[fetchCIDsAndAddToRedis] - Failed to add to blacklist: ${e}`
+        `[fetchCIDsAndAddToRedis] - Failed to add track ids, user ids, or explicitly blacklisted segments to blacklist: ${e.message}`
       )
     }
+
+    await BlacklistManager.addAggregateCIDsToRedis([
+      ...allTrackIdsToBlacklistSet
+    ])
+  }
+
+  /**
+   * Helper method to batch adding CIDs to the blacklist
+   * @param {number[]} allTrackIdsToBlacklist aggregate list of track ids to blacklist from explicit track id blacklist and tracks from blacklisted users
+   */
+  static async addAggregateCIDsToRedis(allTrackIdsToBlacklist) {
+    for (
+      let i = 0;
+      i < allTrackIdsToBlacklist.length;
+      i = i + PROCESS_TRACKS_BATCH_SIZE
+    ) {
+      const tracksSlice = allTrackIdsToBlacklist.slice(
+        i,
+        i + PROCESS_TRACKS_BATCH_SIZE
+      )
+
+      await Utils.asyncRetry({
```

Comment:
why async retry here?

Reply:
in case adding segments fails for a slice of tracks, I think itd be a good idea to retry those segments up to the default 5 times

Reply:
for sure, i'm just wondering in what case it'll error. we didn't have a retry before and i can't think of a case where it would actually throw except if the db query fails

Reply:
yeah actually, I think youre right. let's not over optimize, especially since we've never seen this specific flow fail